src/client/options/mod.rs

```diff
@@ -149,22 +149,28 @@ impl StreamAddress {
         let hostname = match parts.next() {
             Some(part) => part,
             None => {
-                return Err(ErrorKind::InvalidHostname {
```

Comment:
It didn't seem super helpful to have a separate error case for this function only, so I lumped it in with the already existing `ArgumentError` case.